caravel/assets/spec/javascripts/explore/components/actions_spec.js

```diff
@@ -1,11 +1,111 @@
 import { it, describe } from 'mocha';
-import { expect } from 'chai';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import nock from 'nock';
+import { expect, assert } from 'chai';
+import sinon from 'sinon';
 import shortid from 'shortid';
 import * as actions from '../../../../javascripts/explorev2/actions/exploreActions';
 import { initialState } from '../../../../javascripts/explorev2/stores/store';
 import { exploreReducer } from '../../../../javascripts/explorev2/reducers/exploreReducer';
 
+const middlewares = [thunk];
+const mockStore = configureMockStore(middlewares);
+
+describe('ajax call for datasource metadata', () => {
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  it('should return a function', () => {
+    expect(actions.setFormOpts(999, 'test')).to.be.function;
+  });
+
+  it('should dispatch clearAllOpts', () => {
+    const dispatch = sinon.spy();
+    actions.setFormOpts(null, null)(dispatch);
+    assert(dispatch.withArgs(actions.clearAllOpts()).calledOnce);
+  });
+
+  it('should dispatch new opts', () => {
+    nock('/caravel')
+      .get('/fetch_datasource_metadata')
+      .query({ datasource_id: 999, datasource_type: 'test' })
+      .reply(200, {
+        datasource_class: 'SqlaTable',
+        time_columns: ['col'],
+        time_grains: [],
+        groupby_cols: [],
+        metrics: [],
+        filter_cols: [],
+      });
+
+    const store = mockStore(initialState);
+    store.dispatch = sinon.spy();
+    store.dispatch(actions.setFormOpts(999, 'test'));
+    expect(store.dispatch.callCount).to.equal(5);
```

Comment:
I tried it but it says 'Cannot read property 'then' of undefined'. It seems that the ajax call never returned.